src/store/modules/recipeStore.js

```diff
@@ -1,99 +1,111 @@
+    RESET_HASCLAPPED(state) {
+      state.hasClapped = 0;
+    },
+  },
+  actions: {
+    addClap(context, payload) {
+      let url = process.env.BASE_URL + "add-clap";
+
+      this.state.api.client
+        .post(
+          url,
+          {
+            recipe_id: payload.recipeId,
+          },
+          this.state.api.options
+        )
+        .then(function (response) {
+          if (response.data.updated) {
+            context.commit("INCREMENT_CLAP", response.data.claps);
+          }
+        })
+        .catch(function (error) {
+          // console.log('clapping error', error.response)
+        });
+    },
+    async fetch_recipe(context, recipeId) {
+      context.commit("SET_LOADING_STATE", true);
+
+      const uri = this.state.named_urls.recipe_resources + "/" + recipeId;
+
+      await this.state.api.client
+        .get(uri, this.state.api.options)
+        .then(function (response) {
+          response.data.ingredients = JSON.parse(response.data.ingredients);
+
+          const parsedData = JSON.parse(response.data.nutritional_detail);
+          const detail = {
+            cal: parsedData.cal,
+            carbs: parsedData.carbs,
+            protein: parsedData.protein,
+            fat: parsedData.fat,
+          };
+          response.data.nutritional_detail = detail;
 
-			this.state.api.client.post(url, {
-				recipe_id: payload.recipeId,
-			}, this.state.api.options)
-				.then(function (response) {
-					if (response.data.updated) {
-						context.commit('INCREMENT_CLAP', response.data.claps)
-					}
-				}).catch(function (error) {
-					context.commit('HANDLE_ERROR', error.response);
-				})
-		},
-		fetch_recipe(context, recipeId) {
-			context.commit("SET_LOADING_STATE", true)
+          context.commit("UPDATE_RECIPE_STATE", response.data);
+          context.commit("SET_LOADING_STATE", false);
+        })
+        .catch(function (error) {
+          // console.log('fetch error', error)
+          context.commit("SET_LOADING_STATE", false);
+        });
+    },
+    reset_hasClapped(context) {
+      context.commit("RESET_HASCLAPPED");
+    },
 
-			let uri = (process.env.NODE_ENV !== "production") ? 'http://localhost:8080/api/v1/recipes/' + recipeId : process.env.BASE_URL + 'recipes/' + recipeId;
+    async post_recipe(context, payload) {
+      let url = process.env.BASE_URL + "recipes";
+      try {
+        const response = await this.state.api.client.post(
+          url,
+          {
+            cuisine: payload.title,
+            description: payload.recipeDescription,
+            imgUrl: payload.imagePath,
+            ingredients: payload.ingredients,
+            is_draft: payload.draft,
+            name: payload.title,
+            nationality: payload.nationality,
+            summary: payload.keywords,
+            cookbook_id: payload.cookbook_id,
+            tags: payload.tags,
+          },
+          {
+            headers: { Authorization: `Bearer ${process.env.DEV_TOKEN}` },
```

Comment:
use axios success and failure handlers instead of a try catch block, also this call does not have to be async